tests/test-multiple-categories/test.yml

```diff
@@ -0,0 +1,8 @@
+category: test
+channels:
+  - conda-forge
+platforms:
+  - linux-64
+
+dependencies:
+  - pyspark =3.4
```

Comment:
I'm trying to test the following with this setup.

- Pandas should be in the lockfile with 2 optional categories, dev and test. It is a source dependency in file `dev.yml`, so we can make sure that special case works.
- PyArrow should be in the lockfile with 2 optional categories, dev and test. It is a subdependency of astropy and PySpark.
- Python should be in the lockfile, with just the main category. It is a dependency of every other package, so this can test `_truncate_main_category`